### PR TITLE
Improve genesis block creation utility provided by "testing" in framework - Closes #6258

### LINKED ISSUE #6258: Improve genesis block creation utility provided by "testing" in framework
### Description

* The genesis block creation utility should be able to be used to replace fixtures in functional & integration tests with ease.

```
createGenesisBlock = <T = AccountDefaultProps>(
	params: CreateGenesisBlock<T>,
): { genesisBlock: GenesisBlock<T>; genesisBlockJSON: Record<string, unknown> }
```
### Motivation

* There are 2 different functions `createGenesisBlock` & `createGenesisBlockWithAccounts` exposed from utils & fixtures respectively to create genesis block which can be confusing.
* Genesis block returned by `createGenesisBlock` is not compatible as input to `ApplicationEnv`
* Genesis block returned by `createGenesisBlockWithAccounts` is not compatible with existing `config` fixtures in tests
* There is no simple way to start a new application instance with forging enabled using the testing utility

### Acceptance Criteria

* There should be a single function exposed from `testing` utility to create genesis block
* The created genesis block should be compatible as input parameter to other testing utility functions
* It should be possible to have default values for commonly used account & delegate information in the genesis block
* Update existing usage(s) of `createGenesisBlock` & `createGenesisBlockWithAccounts` in tests to the new changes

### Additional Information

* This issue is applicable for branch `feature/6115-add-application-testing-framework`

* To reproduce error `Type 'GenesisBlock<AccountDefaultProps>' is not assignable to type 'Record<string, unknown>'`
```
		const modules = [TokenModule, SequenceModule, KeysModule, DPoSModule];
		const { genesisBlockJSON } = testing.createGenesisBlock(modules);
		config.label = 'reportMisbehavior';
		appEnv = new testing.ApplicationEnv({
			modules,
			config,
			plugins: [ReportMisbehaviorPlugin],
			genesisBlock: genesisBlockJSON,
		});
```
* To reproduce error `'Genesis block does not match'` use following setup with forger plugin functional tests
```
		const modules = [TokenModule, SequenceModule, KeysModule, DPoSModule];
		const { genesisBlockJSON } = testing.createGenesisBlockWithAccounts(modules);
		config.label = 'forging_info_spec';
		appEnv = new testing.ApplicationEnv({
			modules,
			config,
			plugins: [ForgerPlugin],
			genesisBlock: genesisBlockJSON,
		});
		await appEnv.startApplication();
		await appEnv.waitNBlocks(2);
```
* List & replace all existing fixtures for genesis block across tests in lisk-sdk with testing utils, create a separate issue to cleanup fixtures

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #6258 

### How was it solved?

- Remove unwanted fixtures and update tests and utils
- createGenesisBlock util will have default values except modules

### How was it tested?

- Updated tests

### CODE REVIEW

File: framework/src/testing/create_genesis_block.ts [hunk 1]
```diff
@@ -32,22 +35,52 @@ interface CreateGenesisBlock<T> {
 
 export const createGenesisBlock = <T = AccountDefaultProps>(
 	params: CreateGenesisBlock<T>,
-): GenesisBlock<T> => {
-	const accountAssetSchemas = getAccountSchemaFromModules(params.modules, params.genesisConfig);
+): { genesisBlock: GenesisBlock<T>; genesisBlockJSON: Record<string, unknown> } => {
+	// TODO: Remove this dependency in future
+	const modules = [...params.modules];
+	if (!params.modules.includes(DPoSModule)) {
```
Comment (shuse2) [inline framework/src/testing/create_genesis_block.ts:41]: why do we need this?

**Review by shuse2 — APPROVED** [review]

File: framework/src/testing/app_env.ts [hunk 2]
```diff
@@ -104,11 +103,7 @@ export class ApplicationEnv {
 		// As we can call this function with different configuration
 		// so we need to make sure existing schemas are already clear
 		codec.clearCache();
-		// TODO: Remove this dependency in future
-		if (!appConfig.modules.includes(DPoSModule)) {
-			appConfig.modules.push(DPoSModule);
-		}
```
Comment (Incede) [inline framework/src/testing/app_env.ts:110]: So now with this change, if someone initializes AppEnv with empty modules `[]` as input and then start application with `AppEnv.startApplication()` how will the application start (and/or forge blocks) without `DPos` module ?

File: framework/src/testing/fixtures/accounts.ts [hunk 3]
```diff
@@ -13,136 +13,66 @@
...
-	password: 'elephant tree paris dragon chair galaxy',
+	password: defaultPassword,
+};
+
+export const defaultAccounts = <T>(): PartialAccount<T>[] =>
+	defaultConfig.forging.delegates.map(
+		account => ({ address: Buffer.from(account.address, 'hex') } as PartialAccount<T>),
+	);
+
+const getDelegateFromDefaultConfig = (address: Buffer) => {
```
Comment (Incede) [inline framework/src/testing/fixtures/accounts.ts:44]: I am not sure if this function should be in `fixtures`. maybe move it to `utils`.
Same for other functions below.

**Review by Incede — CHANGES_REQUESTED** [review]

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/utils/application.ts [hunk 4]
```diff
@@ -37,21 +40,56 @@ export const createApplicationEnv = (
...
+			force: true,
+		},
 		rpc: {
 			enable: true,
 			port: 8080,
 			mode: 'ipc',
 		},
 	} as PartialApplicationConfig;
 
+	const dataPath = join(rootPath.replace('~', homedir()), label);
```
Comment (shuse2) [inline framework-plugins/lisk-framework-http-api-plugin/test/functional/utils/application.ts:57]: is this handling correct? homedir/label is not the data path

File: framework-plugins/lisk-framework-http-api-plugin/test/functional/utils/application.ts [hunk 5]
```diff
@@ -37,21 +40,56 @@ export const createApplicationEnv = (
 			logFileName: 'lisk.log',
 		},
 		network: {
-			...configJSON.network,
+			...testing.fixtures.defaultConfig.network,
 			maxInboundConnections: 0,
```
Comment (shuse2) [inline framework-plugins/lisk-framework-http-api-plugin/test/functional/utils/application.ts:44]: maybe `maxInboundConnection: 0` can be default, and rpc, forging settings

**Review by Incede — APPROVED** [review]

**Pull request merged**